src/lib/electron.ts

```diff
@@ -0,0 +1,501 @@
+      sdk: { name: SDK_NAME, version: SDK_VERSION },
+      breadcrumbs: this.breadcrumbs.get(),
+    };
+
+    return this.callInner(inner => inner.send(mergedEvent));
+  }
+
+  /**
+   * Updates options of this SDK.
+   *
+   * Note that once crash reporting is enabled for either JavaScript errors or
+   * native crashes, they cannot be disabled anymore.
+   *
+   * @param options New options.
+   */
+  public async setOptions(options: SentryElectronOptions): Promise<void> {
+    this.options = options;
+
+    if (this.isNativeEnabled()) {
+      // We can safely re-install the native CrashReporter. If it had been
+      // started before, this will effectively be a no-op. Otherwise, it will
+      // start a new instance.
+      await this.installNativeHandler();
+    } else {
+      this.client.log('Native CrashReporter cannot be stopped once started.');
+    }
+
+    return this.callInner(inner => inner.setOptions(options));
+  }
+
+  /**
+   * Returns the context saved for this app.
+   *
+   * After a native crash, this tries to recover saved context from disk. Use
+   * {@link setContext} to change this context. Note that the context is managed
+   * by the main process, so calling this method in the renderer process will
+   * not return any data.
+   *
+   * @returns The current context.
+   */
+  public async getContext(): Promise<Context> {
+    // The context is managed by the main process only
+    return this.isMainProcess() ? this.context.get() : {};
+  }
+
+  /**
+   * Sets a new context for this app.
+   *
+   * The context is managed by the main process. Calling this method in a
+   * renderer process will issue an IPC message to the main process. The context
+   * is asynchronously flushed to disk so that it can be recovered when the
+   * application crashes.
+   *
+   * @param context A new context to replace the previous one.
+   */
+  public async setContext(context: Context): Promise<void> {
+    if (this.isRenderProcess()) {
+      ipcRenderer.send(IPC_CONTEXT, context);
+      return;
+    }
+
+    this.context.set(context);
+  }
+
+  /** Returns whether the SDK is running in the main process. */
+  private isMainProcess(): boolean {
+    return process.type === 'browser';
+  }
+
+  /** Returns whether the SDK is running in a renderer process. */
+  private isRenderProcess(): boolean {
+    return process.type === 'renderer';
+  }
+
+  /** Returns whether JS is enabled and we are in the main process. */
+  private isMainEnabled(): boolean {
+    return this.isMainProcess() && this.options.enableJavaScript !== false;
+  }
+
+  /** Returns whether JS is enabled and we are in a renderer process. */
+  private isRenderEnabled(): boolean {
+    return this.isRenderProcess() && this.options.enableJavaScript !== false;
+  }
+
+  /** Returns whether native reports are enabled. */
+  private isNativeEnabled(): boolean {
+    // On macOS, we should only start the Electron CrashReporter in the main
```

Comment:
The electron docs seem to contradict this:

>Note: On macOS, Electron uses a new crashpad client for crash collection and reporting. If you want to enable crash reporting, initializing crashpad from the main process using crashReporter.start is required regardless of which process you want to collect crashes from. Once initialized this way, the crashpad handler collects crashes from all processes.

Reply:
Or is that simply so the crashes are sent to the server?

Reply:
Crashpad spawns a new process and attaches to the Electron processes to watch them (required since macOS Mavericks). This is why it should only be necessary to start the crash reporter in the main process. It seems that this is also what the Electron docs try to express. 
On Windows and Linux, Electron still uses Breakpad. In that case, it should be necessary to start the CrashReporter in every process. 
So far, I have only tested thoroughly on macOS.